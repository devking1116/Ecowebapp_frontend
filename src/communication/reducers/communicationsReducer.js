/* @flow */
import { COMMUNICATION_REDUCER_LOAD } from '../actions/loadCommunicationsAction';
import { COMMUNICATION_REDUCER_SELECT } from '../actions/selectCommunicationAction';

export type Communication = {
  title: string,
  author: string,
  date: string,
  status?: 'active' | 'in-active',
  attachments?: Array<{
    id: number,
    fileName: string,
    size: string,
  }>,
  comments?: Array<{
    author: string,
    date: string,
    status?: 'active' | 'in-active',
    actionsTakens: Array<string>,
    linkedLogbooks: Array<string>,
    attachments?: Array<number>,
  }>,
  activity?: Array<{
    author: string,
    label: string,
    date: string,
  }>,
};

type State = {
  communications: Array<Communication>,
  selectedCommunication: ?Communication,
};

const initialState = {
  communications: [],
  selectedCommunication: null,
};

/**
 * The reducer that contains all the communication data for the site.
 */
export const COMMUNICATION_REDUCER_STATE = 'COMMUNICATION_REDUCER_STATE';
export default function communicationsReducer(state: State = initialState, action: Object) {
  switch (action.type) {
    case COMMUNICATION_REDUCER_LOAD:
      //TODO, replace hard-coded data with data from the api
      return {
        ...state,
        communications: [
          { title: 'Testing departments acess', author: 'User 0', date: '14/Dec/2016 00:21 PM MST', status: 'active' },
          { title: 'Test communication 1', author: 'User 1', date: '14/Dec/2016 01:21 PM MST' },
          { title: 'Test communication 2', author: 'User 2', date: '14/Dec/2016 01:21 PM MST', status: 'in-active' },
        ],
      };
    case COMMUNICATION_REDUCER_SELECT:
      //TODO, replace hard-coded data with data from the api
      return {
        ...state,
        selectedCommunication: {
          ...state.communications[action.index],
          attachments: [
            { id: 0, fileName: 'testAttachment0.pdf', modifiedDate: 'January 3, 2017', size: '67.0kb' },
            { id: 1, fileName: 'testAttachment1.pdf', modifiedDate: 'January 4, 2016', size: '67.1kb' },
            { id: 2, fileName: 'testAttachment2.pdf', modifiedDate: 'January 7, 2017', size: '67.2kb' },
          ],
          comments: [
            {
              author: 'Richard Simms',
              date: 'Today 5:45 am',
              body: 'Test comment 0',
              status: 'active',
              actionsTakens: ['Spoke to Resident', 'Jetted Sweer Line'],
              linkedLogbooks: [ 'logbook A' ],
            },
            {
              author: 'Randy S.Beard',
              date: '1/1/2016',
              body: 'Test comment 1',
              linkedLogbooks: [ 'logbook A', 'logbook B' ],
              attachments: [ 0 ],
            },
            {
              author: 'User 2',
              date: '2/2/2016',
              body: 'Test comment 2',
            },
            {
              author: 'User 3',
              date: '3/3/2016',
              body: 'Test comment 3',
              status: 'in-active',
              attachments: [ 1, 2 ],
            },
          ],
          activity: [
            { author: 'User 0', date: '14/Dec/2016 00:21 PM MST', label: 'commented on this post' },
            { author: 'User 1', date: '14/Dec/2016 01:21 PM MST', label: 'added two attachments' },
            { author: 'User 2', date: '14/Dec/2016 02:21 PM MST', label: 'change status to active' },
          ],
        },
      };
    default:
      return state;
  }
}
