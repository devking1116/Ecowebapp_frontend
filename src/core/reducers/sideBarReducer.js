/* @flow */
import { SIDEBAR_COLLAPSE } from '../actions/collapseSideBarAction';
import { SIDEBAR_EXPAND } from '../actions/expandSideBarAction';
import { SIDEBAR_REGISTER } from '../actions/registerSideBarAction';
import { SIDEBAR_UNREGISTER } from '../actions/unregisterSideBarAction';

export type SideBarState = {
  expandedWidth: number | string,
  isExpanded: bool,
};

export type State = { [key: string]: SideBarState };

/**
 * A reducer to control weather or not a component is expaned or collapsed.
 */
export const SIDEBAR_REDUCER_STATE = 'SIDEBAR_REDUCER_STATE';
export default function sideBarReducer(state: State = { }, action: Object): State {
  switch(action.type) {
    case SIDEBAR_REGISTER:
      if(!action.sideBarId) {
        throw new Error('Side Bar must be given to unregister');
      }
      if(typeof action.expandedWidth !== 'number' && typeof action.expandedWidth !== 'string') {
        throw new Error('Side Bar expandedWidth must be number or string');
      }

      return {
        ...state,
        [action.sideBarId]: {
          expandedWidth: action.expandedWidth,
          isExpanded: action.defaultExpanded? true: false,
        },
      };
    case SIDEBAR_UNREGISTER:
      if(!action.sideBarId)         throw new Error('Side Bar must be given to unregister');

      return {
        ...state,
        [action.sideBarId]: undefined,
      };
    case SIDEBAR_EXPAND:
    case SIDEBAR_COLLAPSE:
      if(!action.sideBarId)         throw new Error('Side Bar must be given to update');
      if(!state[action.sideBarId])  throw new Error('Side Bar must be registered to update');

      return {
        ...state,
        [action.sideBarId]: {
          ...state[action.sideBarId],
          isExpanded: action.type === SIDEBAR_EXPAND? true: false,
        },
      };
    default:
      return state;
  }
}
