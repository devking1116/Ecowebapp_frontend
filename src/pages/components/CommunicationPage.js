/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Icon, Popup } from 'semantic-ui-react';

import { Communication, CommunicationNavBar, CommunicationSearchResult } from '../../communication';
import { FillRemaining, SearchSideBar, SideBarNavGroup, SideBarNavItem } from '../../core';

import type { Communication as CommunicationData } from '../../communication/reducers/communicationsReducer';

type Props = {
  results: Array<CommunicationData>,
  children?: any,
};

const COMMUNICATION_NAV_BAR = 'COMMUNICATION_NAV_BAR';
const COMMUNICATION_SEARCH_BAR = 'COMMUNICATION_SEARCH_BAR';

/**
 * Wrapper for the Communication Page, that contains the side bar and panel.
 */
export default function CommunicationPage({ results, children, ...otherProps }: Props): React.Element<*> {
  return (
    <FillRemaining direction="row" {...otherProps}>
      <FillRemaining direction="row">
        <CommunicationNavBar sideBarId={COMMUNICATION_NAV_BAR} initialExpandedWidth={200}>
          <SideBarNavGroup>
            <SideBarNavItem label="Customer Calls" icon={<Icon name="call" size="large" />} />
            <SideBarNavItem label="Daily Bulletins" icon={<Icon name="file text" size="large" />} />
            <SideBarNavItem label="Discussions" icon={<Icon name="discussions" size="large" />} />
            <SideBarNavItem label="Emergency Alerts" icon={<Icon name="exclamation triangle" size="large" />} />
          </SideBarNavGroup>
          <SideBarNavGroup>
            <SideBarNavItem label="System Manager" icon={<Icon name="setting" size="large" />} />
            <SideBarNavItem label="Archives" icon={<Icon name="archive" size="large" />} />
          </SideBarNavGroup>
        </CommunicationNavBar>
        <SearchSideBar sideBarId={COMMUNICATION_SEARCH_BAR} initialExpandedWidth={400} defaultExpanded>
        {results.map((result, index) =>
          <CommunicationSearchResult {...result} key={index} />
        )}
        </SearchSideBar>
      </FillRemaining>
      <Communication>
        {children}
      </Communication>
    </FillRemaining>
  );
}
