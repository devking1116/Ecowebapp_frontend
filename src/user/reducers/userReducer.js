/* @flow */
import { USER_REDUCER_LOGIN } from '../actions/loginUserAction';

type State = {
  clientName: ?string,
  clientId: ?string,
  isAdmin: bool,
};

const initialState = {
  clientName: null,
  clientId: null,
  isAdmin: false,
};

/**
 * A reducer to keep track of the data for the logged in user.
 */
export const USER_REDUCER_STATE = 'USER_REDUCER_STATE';
export default function userReducer(state: State = initialState, action: *): State {
  switch (action.type) {
    case USER_REDUCER_LOGIN:
      //TODO, replace hard-coded data with data from the api
      return {
        clientName: 'Client Name',
        clientId:'123456',
        isAdmin: true,
      };
    default:
      return state;
  }
}
