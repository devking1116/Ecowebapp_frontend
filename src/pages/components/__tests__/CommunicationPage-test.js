import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import CommunicationPage from '../CommunicationPage';
import { COMMUNICATION_REDUCER_STATE } from '../../../communication/reducers/communicationsReducer';
import { SIDEBAR_REDUCER_STATE } from '../../../core/reducers/sideBarReducer';

function makeMockStore() {
  const getNewStore = configureStore([]);

  return getNewStore({
    [COMMUNICATION_REDUCER_STATE]: { },
    [SIDEBAR_REDUCER_STATE]: { },
  });
}

test('it displays the children next to the sidebar.', () => {
  const component = renderer.create(
    <Provider store={makeMockStore()}>
      <CommunicationPage results={[]}>
        Test Communication Page Content
      </CommunicationPage>
    </Provider>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays the given results in the sidebar.', () => {
  const component = renderer.create(
    <Provider store={makeMockStore()}>
      <CommunicationPage
        results={[
          { title: 'Testing department access', author: 'User 0', date: '14/Dec/2016 00:21 PM MST', status: 'active' },
          { title: 'Test communication 1', author: 'User 1', date: '14/Dec/2016 01:21 PM MST' },
          { title: 'Test communication 2', author: 'User 2', date: '14/Dec/2016 01:21 PM MST', status: 'in-active' },
        ]}
      >
        Test Communication Page Content
      </CommunicationPage>
    </Provider>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
