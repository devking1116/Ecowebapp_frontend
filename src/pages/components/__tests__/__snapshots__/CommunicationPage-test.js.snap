exports[`test it displays the children next to the sidebar. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "width": "100%",
    }
  }>
  <div
    style={
      Object {
        "height": "100%",
      }
    }>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": "100%",
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "height": "100%",
          }
        }>
        <div
          className="ui inverted vertical icon menu"
          style={
            Object {
              "MozBorderRadius": 0,
              "OBorderRadius": 0,
              "WebkitBorderRadius": 0,
              "backgroundColor": "#2b5f8c",
              "borderRadius": 0,
              "display": undefined,
              "height": "100%",
              "msBorderRadius": 0,
              "position": "relative",
              "width": undefined,
            }
          }>
          <div
            className="item"
            onClick={[Function]}>
            <div
              style={
                Object {
                  "WebkitJustifyContent": "flex-end",
                  "display": "flex",
                  "height": 20,
                  "justifyContent": "flex-end",
                }
              }>
              <i
                aria-hidden="true"
                className="content large icon"
                onClick={[Function]} />
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}>
            <div
              className="menu">
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="call large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="file text large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="discussions large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="exclamation triangle large icon" />
                </span>
              </div>
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}>
            <div
              className="menu">
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="setting large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="archive large icon" />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "MozBoxFlex": "1 0 1px",
            "WebkitBoxFlex": "1 0 1px",
            "WebkitFlex": "1 0 1px",
            "flex": "1 0 1px",
            "height": "100%",
            "msFlex": "1 0 1px",
          }
        }>
        <div
          className="ui vertical icon menu"
          style={
            Object {
              "backgroundColor": "#f8f8f8",
              "display": undefined,
              "height": "100%",
              "position": "relative",
              "width": undefined,
            }
          }>
          <div
            className="item"
            onClick={[Function]}>
            <div
              style={
                Object {
                  "WebkitJustifyContent": "flex-end",
                  "display": "flex",
                  "height": 20,
                  "justifyContent": "flex-end",
                }
              }>
              <i
                aria-hidden="true"
                className="content large icon"
                onClick={[Function]} />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "MozBoxFlex": "1 0 1px",
        "WebkitBoxFlex": "1 0 1px",
        "WebkitFlex": "1 0 1px",
        "flex": "1 0 1px",
        "height": "100%",
        "msFlex": "1 0 1px",
      }
    }>
    <section
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "width": "100%",
          }
        }>
        <div
          style={Object {}}>
          <header
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "paddingTop": 15,
                "position": "relative",
              }
            }>
            <div
              style={
                Object {
                  "display": "inline-flex",
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                }
              }>
              <div
                className="ui left icon input"
                style={
                  Object {
                    "width": 400,
                  }
                }>
                <i
                  aria-hidden="true"
                  className="search icon" />
                <input
                  type="text" />
              </div>
              <div
                style={
                  Object {
                    "marginLeft": -50,
                    "paddingTop": 10,
                    "zIndex": 1001,
                  }
                }>
                <a>
                  <i
                    aria-hidden="true"
                    className="close icon" />
                </a>
                <a>
                  <i
                    aria-hidden="true"
                    className="black caret down icon" />
                </a>
              </div>
              <button
                className="ui primary button"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": 20,
                    "width": 100,
                  }
                }
                tabIndex={undefined}>
                Search
              </button>
            </div>
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "width": "60%",
                  }
                }>
                <h3
                  style={
                    Object {
                      "marginBottom": 0,
                    }
                  }>
                  
                </h3>
                <span>
                  In 
                  <a>
                    Operations
                  </a>
                   > 
                  <a>
                    Daily Check-sheet
                  </a>
                   | Department Admionistration
                </span>
              </div>
              <div
                aria-busy={undefined}
                aria-disabled={undefined}
                aria-expanded={false}
                aria-multiselectable={undefined}
                className="ui selection dropdown"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                role="listbox"
                style={
                  Object {
                    "backgroundColor": "green",
                    "color": "white",
                    "float": "right",
                    "marginRight": 15,
                    "marginTop": -35,
                  }
                }
                tabIndex={0}>
                <select
                  aria-hidden="true"
                  multiple={undefined}
                  name={undefined}
                  type="hidden"
                  value="">
                  <option
                    value="" />
                  <option
                    value="active">
                    Active
                  </option>
                  <option
                    value="in-active">
                    In-Active
                  </option>
                  <option
                    value="none">
                    None
                  </option>
                </select>
                <div
                  className="default text">
                  Status
                </div>
                <i
                  aria-hidden="true"
                  className="dropdown icon" />
                <div
                  className="menu transition">
                  <div
                    aria-checked={false}
                    aria-disabled={undefined}
                    aria-selected={true}
                    className="selected item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }>
                    Active
                  </div>
                  <div
                    aria-checked={false}
                    aria-disabled={undefined}
                    aria-selected={false}
                    className="item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }>
                    In-Active
                  </div>
                  <div
                    aria-checked={false}
                    aria-disabled={undefined}
                    aria-selected={false}
                    className="item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }>
                    None
                  </div>
                </div>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "relative",
                }
              }>
              <div
                className="ui secondary menu">
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="commenting icon" />
                  <span>
                    Comment
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="tag icon" />
                  <span>
                    Logbook
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="share icon" />
                  <span>
                    Share
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="edit icon" />
                  <span>
                    Edit
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="trash outline icon" />
                  <span>
                    Delete
                  </span>
                </div>
              </div>
              <div
                className="ui tabular menu"
                style={
                  Object {
                    "paddingLeft": 10,
                  }
                }>
                <div
                  className="active item"
                  onClick={[Function]}>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    Details
                  </a>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    Comments
                  </a>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    Activity
                  </a>
                </div>
              </div>
            </div>
          </header>
        </div>
        <div
          style={
            Object {
              "MozBoxFlex": "1 0 1px",
              "WebkitBoxFlex": "1 0 1px",
              "WebkitFlex": "1 0 1px",
              "display": "flex",
              "flex": "1 0 1px",
              "flexDirection": "row",
              "msFlex": "1 0 1px",
            }
          }>
          <div
            style={
              Object {
                "position": "relative",
                "width": "100%",
              }
            }>
            <div
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }>
              <div
                style={
                  Object {
                    "height": "100%",
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }>
                Test Communication Page Content
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`test it displays the given results in the sidebar. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "width": "100%",
    }
  }>
  <div
    style={
      Object {
        "height": "100%",
      }
    }>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": "100%",
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "height": "100%",
          }
        }>
        <div
          className="ui inverted vertical icon menu"
          style={
            Object {
              "MozBorderRadius": 0,
              "OBorderRadius": 0,
              "WebkitBorderRadius": 0,
              "backgroundColor": "#2b5f8c",
              "borderRadius": 0,
              "display": undefined,
              "height": "100%",
              "msBorderRadius": 0,
              "position": "relative",
              "width": undefined,
            }
          }>
          <div
            className="item"
            onClick={[Function]}>
            <div
              style={
                Object {
                  "WebkitJustifyContent": "flex-end",
                  "display": "flex",
                  "height": 20,
                  "justifyContent": "flex-end",
                }
              }>
              <i
                aria-hidden="true"
                className="content large icon"
                onClick={[Function]} />
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}>
            <div
              className="menu">
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="call large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="file text large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="discussions large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="exclamation triangle large icon" />
                </span>
              </div>
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}>
            <div
              className="menu">
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="setting large icon" />
                </span>
              </div>
              <div
                className="item"
                onClick={[Function]}>
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}>
                  <i
                    aria-hidden="true"
                    className="archive large icon" />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "MozBoxFlex": "1 0 1px",
            "WebkitBoxFlex": "1 0 1px",
            "WebkitFlex": "1 0 1px",
            "flex": "1 0 1px",
            "height": "100%",
            "msFlex": "1 0 1px",
          }
        }>
        <div
          className="ui vertical icon menu"
          style={
            Object {
              "backgroundColor": "#f8f8f8",
              "display": undefined,
              "height": "100%",
              "position": "relative",
              "width": undefined,
            }
          }>
          <div
            className="item"
            onClick={[Function]}>
            <div
              style={
                Object {
                  "WebkitJustifyContent": "flex-end",
                  "display": "flex",
                  "height": 20,
                  "justifyContent": "flex-end",
                }
              }>
              <i
                aria-hidden="true"
                className="content large icon"
                onClick={[Function]} />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "MozBoxFlex": "1 0 1px",
        "WebkitBoxFlex": "1 0 1px",
        "WebkitFlex": "1 0 1px",
        "flex": "1 0 1px",
        "height": "100%",
        "msFlex": "1 0 1px",
      }
    }>
    <section
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "width": "100%",
          }
        }>
        <div
          style={Object {}}>
          <header
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "paddingTop": 15,
                "position": "relative",
              }
            }>
            <div
              style={
                Object {
                  "display": "inline-flex",
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                }
              }>
              <div
                className="ui left icon input"
                style={
                  Object {
                    "width": 400,
                  }
                }>
                <i
                  aria-hidden="true"
                  className="search icon" />
                <input
                  type="text" />
              </div>
              <div
                style={
                  Object {
                    "marginLeft": -50,
                    "paddingTop": 10,
                    "zIndex": 1001,
                  }
                }>
                <a>
                  <i
                    aria-hidden="true"
                    className="close icon" />
                </a>
                <a>
                  <i
                    aria-hidden="true"
                    className="black caret down icon" />
                </a>
              </div>
              <button
                className="ui primary button"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": 20,
                    "width": 100,
                  }
                }
                tabIndex={undefined}>
                Search
              </button>
            </div>
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "width": "60%",
                  }
                }>
                <h3
                  style={
                    Object {
                      "marginBottom": 0,
                    }
                  }>
                  
                </h3>
                <span>
                  In 
                  <a>
                    Operations
                  </a>
                   > 
                  <a>
                    Daily Check-sheet
                  </a>
                   | Department Admionistration
                </span>
              </div>
              <div
                aria-busy={undefined}
                aria-disabled={undefined}
                aria-expanded={false}
                aria-multiselectable={undefined}
                className="ui selection dropdown"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                role="listbox"
                style={
                  Object {
                    "backgroundColor": "green",
                    "color": "white",
                    "float": "right",
                    "marginRight": 15,
                    "marginTop": -35,
                  }
                }
                tabIndex={0}>
                <select
                  aria-hidden="true"
                  multiple={undefined}
                  name={undefined}
                  type="hidden"
                  value="">
                  <option
                    value="" />
                  <option
                    value="active">
                    Active
                  </option>
                  <option
                    value="in-active">
                    In-Active
                  </option>
                  <option
                    value="none">
                    None
                  </option>
                </select>
                <div
                  className="default text">
                  Status
                </div>
                <i
                  aria-hidden="true"
                  className="dropdown icon" />
                <div
                  className="menu transition">
                  <div
                    aria-checked={false}
                    aria-disabled={undefined}
                    aria-selected={true}
                    className="selected item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }>
                    Active
                  </div>
                  <div
                    aria-checked={false}
                    aria-disabled={undefined}
                    aria-selected={false}
                    className="item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }>
                    In-Active
                  </div>
                  <div
                    aria-checked={false}
                    aria-disabled={undefined}
                    aria-selected={false}
                    className="item"
                    onClick={[Function]}
                    role="option"
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }>
                    None
                  </div>
                </div>
              </div>
            </div>
            <div
              style={
                Object {
                  "position": "relative",
                }
              }>
              <div
                className="ui secondary menu">
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="commenting icon" />
                  <span>
                    Comment
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="tag icon" />
                  <span>
                    Logbook
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="share icon" />
                  <span>
                    Share
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="edit icon" />
                  <span>
                    Edit
                  </span>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <i
                    aria-hidden="true"
                    className="trash outline icon" />
                  <span>
                    Delete
                  </span>
                </div>
              </div>
              <div
                className="ui tabular menu"
                style={
                  Object {
                    "paddingLeft": 10,
                  }
                }>
                <div
                  className="active item"
                  onClick={[Function]}>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    Details
                  </a>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    Comments
                  </a>
                </div>
                <div
                  className="item"
                  onClick={[Function]}>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    Activity
                  </a>
                </div>
              </div>
            </div>
          </header>
        </div>
        <div
          style={
            Object {
              "MozBoxFlex": "1 0 1px",
              "WebkitBoxFlex": "1 0 1px",
              "WebkitFlex": "1 0 1px",
              "display": "flex",
              "flex": "1 0 1px",
              "flexDirection": "row",
              "msFlex": "1 0 1px",
            }
          }>
          <div
            style={
              Object {
                "position": "relative",
                "width": "100%",
              }
            }>
            <div
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }>
              <div
                style={
                  Object {
                    "height": "100%",
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }>
                Test Communication Page Content
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
